src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -808,18 +832,32 @@ public bool GrowAllocation(AllocatedMemoryData allocation, int sizeIncrease)
         }
 
 
+        public bool TryReturnMemory(AllocatedMemoryData allocation)
```

Comment:
What _is_ this? You are guaranteeing a leak here, no? The use after free is there for a reason.
Just ignoring it means that we are opening ourselves to a LOT of issues.